org.eclipse.winery.repository.ui/src/app/instance/instanceHeader/propertyRename/propertyRename.component.ts

```diff
@@ -0,0 +1,105 @@
+ */
+
+import { Component, Input, OnInit, ViewChild } from '@angular/core';
+import { WineryNotificationService } from '../../../wineryNotificationModule/wineryNotification.service';
+import { PropertyRenameService } from './propertyRename.service';
+import { ToscaComponent } from '../../../wineryInterfaces/toscaComponent';
+import { SectionData } from '../../../section/sectionData';
+import { WineryValidatorObject } from '../../../wineryValidators/wineryDuplicateValidator.directive';
+import { NgForm } from '@angular/forms';
+import { Router } from '@angular/router';
+
+/**
+ * This adds a an editable field to the html that manipulates either the namespace or the id/name of a ToscaComponent
+ *
+ * @Input id: this is the id of the property which must either be 'id' or 'namespace'
+ * @Input propertyName: the name of the property in the backend, can be 'id' or 'ns'
+ * @Input toscaComponent: the toscaComponent which's id/namespace is edited/displayed
+ * @Input currentValue: the value that should be initially displayed this will also be used for the initial edit value
+ */
+@Component({
+    selector: 'winery-property-rename',
+    templateUrl: 'propertyRename.component.html',
+    providers: [PropertyRenameService],
+    styleUrls: [
+        'propertyRename.component.css'
+    ]
+})
+export class PropertyRenameComponent implements OnInit {
+    @Input() id: string;
+    @Input() propertyName: string;
+    @Input() toscaComponent: ToscaComponent;
+    @Input() currentValue: string;
+    @ViewChild('renameComponentForm') renameComponentForm: NgForm;
+    editMode = false;
+    canEdit = true;
+    propertyValue = '';
+    componentData: SectionData[];
+    validatorObject: WineryValidatorObject;
+
+    constructor(private service: PropertyRenameService,
+                private notify: WineryNotificationService,
+                private router: Router) {
+    }
+
+    ngOnInit(): void {
+        this.service.setToscaComponent(this.toscaComponent);
+        this.service.setId(this.id);
+        this.service.setPropertyName(this.propertyName);
+        this.canEdit = this.router.url.includes('admin');
+    }
+
+    updateValue() {
+        this.service.setPropertyValue(this.propertyValue).subscribe(
+            data => this.handleUpdateValue(),
+            error => this.handleError(error)
+        )
+    }
+
+    onClickEdit() {
+        this.validatorObject = new WineryValidatorObject(this.componentData, this.id);
+        this.propertyValue = this.currentValue;
+        this.editMode = true;
+    }
+
+    onSaveValue() {
+        this.editMode = false;
+        this.updateValue();
+    }
+
+    onCancel() {
+        this.editMode = false;
+    }
+
+    private handleUpdateValue() {
+        this.notify.success('Renamed ' + this.id + ' to ' + this.propertyValue);
+        this.service.reload(this.propertyValue);
+    }
+
+    private handleError(error: any): void {
+        this.notify.error('id/name ' + this.propertyValue
```

Comment:
~id~ => localName
~name~ => namespace

Reply:
the official term in the tosca specification see winery\docs\tosca\TOSCA-v1.0-os-attributes.xlsx is either id or name

Reply:
I think, in the context of the project, we decided to use "localName" in the code.

Towards the user, we use "name" and "id".